bin/bulk_create

```diff
@@ -0,0 +1,274 @@
+    default_rmt_uuid = 'ecf431f8-7faa-48a4-a3f1-6d3abb199086'
+
+    parser = argparse.ArgumentParser(description=
+        'Bulk create systems SCC VirtualizationHosts API testing'
+    )
+
+    # SCC max systems to upload per request
+    parser.add_argument(
+        '--systems_per_request', type=int, action='store', default=default_systems_per_request,
+        help=f'The max number of systems to upload per request to the SCC. (Default: {default_systems_per_request})'
+    )
+
+    # SCC RMT UUID
+    parser.add_argument(
+        '--rmt_uuid', type=str, action='store', default=default_rmt_uuid,
+        help=f'The RMT UUID to report to the SCC. (Default: {default_rmt_uuid})'
+    )
+
+    # SCC RMT Name
+    parser.add_argument(
+        '--rmt_name', type=str, action='store', default=default_rmt_name,
+        help=f'The RMT name to report to the SCC. (Default: {default_rmt_name})'
+    )
+
+    # SCC API base URL
+    parser.add_argument(
+        '-a', '--api_base_url', type=str, action='store', default=default_api_base_url,
+        help=f'The SCC API base URL. (Default: {default_api_base_url})'
+    )
+
+    # SCC mirroring credentials username
+    parser.add_argument(
+        '-u', '--username', type=str, action='store',
+        help='The SCC mirroring credentials username to use for bulk creation. Can be specified via SCC_USERNAME env var.'
+    )
+
+    # SCC mirroring credentials password
+    parser.add_argument(
+        '-p', '--password', type=str, action='store',
+        help='The SCC mirroring credentials password to use for bulk creation. Can be specified via SCC_PASSWORD env var.'
+    )
+
+    # input systems file
+    parser.add_argument(
+        '-s', '--systems', type=Path, action='store', default=default_systems,
+        help=f'The input file containing the details of the Libvirt host and VM systems to bulk create. (Default: {default_systems})'
+    )
+
+    # check creds are valid
+    parser.add_argument(
+        '-c', '--check_creds', action='store_true', default=False,
+        help='Verify that the supplied SCC mirroring credentials are valid.'
+    )
+
+    # parse the supplied arguments
+    return parser.parse_args(argv)
+
+
+def main(argv):
+    args = parse_args(argv)
+
+    if args.username is None:
+        args.username = os.environ.get('SCC_USERNAME')
+
+    if args.password is None:
+        args.password = os.environ.get('SCC_PASSWORD')
+
+    if args.username is None:
+        exit('The SCC username must be specified either via the --username option, or via the SCC_USERNAME env var.')
+
+    if args.password is None:
+        exit('The SCC password must be specified either via the --password option, or via the SCC_PASSWORD env var.')
+
+    scc_config = SccConfig(args.username, args.password, args.api_base_url)
+
+    with args.systems.open('r') as fp:
+        system_details = yaml.safe_load(fp)
+
+    # generate uploadable VM systems entries
+    vm_systems = [
```

Comment:
can we name this vmware_systems just to be more specific?

Reply:
This is not just the vmware systems but the entire set of VMs that are running across the set of hypervisor hosts, whether they are VMWare or Libvirt.